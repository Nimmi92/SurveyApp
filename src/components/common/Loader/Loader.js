import React from 'react';
import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`
const StyledLoader = styled.div`
  border: 5px solid #f3f3f3; 
  border-top: 5px solid #3498db; 
  border-radius: 50%;
  width: 50px;
  height: 50px;
  animation: ${spin} 2s linear infinite;
  top: 50%;
  left: 50%;
  margin: -25px;
  position: absolute;
`

const Loader = () => {

  return (
    <div>
     <StyledLoader />
    </div>
  );
};

export default Loader;
