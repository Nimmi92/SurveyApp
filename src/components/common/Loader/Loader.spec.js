import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';
import toJson from "enzyme-to-json";


describe('<Loader />', () => {
  it('should be an loader element', () => {
    const wrapper = shallow(<Loader />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})