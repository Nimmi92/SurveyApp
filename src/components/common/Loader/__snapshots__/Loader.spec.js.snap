// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should be an loader element 1`] = `
<div>
  <ForwardRef />
</div>
`;
